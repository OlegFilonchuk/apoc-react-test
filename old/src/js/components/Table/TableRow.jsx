import React from 'react';
import PropTypes from 'prop-types';

export default function TableRow({ children, className }) {
  return <tr className={className}>{children}</tr>;
}

TableRow.propTypes = {
  /**
   * Row's content
   */
  children: PropTypes.node,

  /**
   * List of custom classes added to component
   */
  className: PropTypes.string
};

TableRow.defaultProps = {
  children: null,
  className: ''
};
