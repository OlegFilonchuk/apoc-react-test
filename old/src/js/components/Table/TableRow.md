## Example of table with TableRow
It uses [TableHeadRow](#tableheadrow) and [Table](#table)

    (
      <Table>
        <TableHeadRow>
          <th>Title</th>
          <th>Title</th>
          <th>Title</th>
        </TableHeadRow>
        <tbody>
          <TableRow>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRow>
          <TableRow>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRow>
          <TableRow>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </TableRow>
        </tbody>
      </Table>
    )
