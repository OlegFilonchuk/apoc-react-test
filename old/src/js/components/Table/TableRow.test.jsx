import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TableRow from './TableRow';

const mountRow = ({ mountMethod = mount, props = {} }) =>
  mountMethod(
    <TableRow {...props}>
      <td>Content</td>
      <td>Content</td>
    </TableRow>,
    {
      attachTo: document.createElement('tbody')
    }
  );

describe('<TableRow />', () => {
  it('should match snapshot', () => {
    expect(toJson(mountRow({ mountMethod: shallow }))).toMatchSnapshot();
  });

  it('correctly set `className`', () => {
    const cell = mountRow({ props: { className: 'some-extra-class' } });
    const element = cell.find('tr');

    expect(element).toHaveClassName('some-extra-class');
  });
});
