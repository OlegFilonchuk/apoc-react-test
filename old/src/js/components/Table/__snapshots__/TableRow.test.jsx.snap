// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow /> should match snapshot 1`] = `
<tr
  className=""
>
  <td>
    Content
  </td>
  <td>
    Content
  </td>
</tr>
`;
